health-services/project/src/main/java/org/egov/project/repository/UserRepository.java

```diff
@@ -0,0 +1,57 @@
+package org.egov.project.repository;
+
+import digit.models.coremodels.UserSearchRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.common.contract.request.User;
+import org.egov.project.web.models.UserServiceResponse;
+import org.egov.tracer.model.CustomException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+@Slf4j
+public class UserRepository {
+
+    private final RestTemplate restTemplate;
+
+    @Value("${egov.user.host}")
+    private String HOST;
+
+    @Value("${egov.search.user.url}")
+    private String SEARCH_USER_URL;
+
+    @Autowired
+    public UserRepository(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public List<String> validatedUserIds(List<String> userIds, String tenantId) {
```

Comment:
should be `validateUserIds` or `searchByUserIds`

Reply:
Updated